packages/components/src/DateTimePickers/LegacyDateTimePickers/DateTime/date-extraction.js

```diff
@@ -460,6 +487,28 @@ function extractPartsFromDateAndTime(date, time, options) {
 	};
 }
 
+function extractDateOrTimeHybridMode(textInput, options) {
+	let date;
```

Comment:
Another way to do that is to have separate function:
- a formatDate (with strToDate(textInput, options.dateFormat) or an error)
- a formatTime (with strToTime(textInput, options.useSeconds); or an error)
- And you're extractDateOrTimeHybridMode that handle this two one and build the response (an object or an error).

This will help you to have a more concise code, simple to read and easy to test